v3iofs/fs.py

```diff
@@ -56,58 +54,70 @@ def __init__(self, v3io_api=None, v3io_access_key=None, **kw):
             transport_kind='requests',
         )
 
-    def ls(self, path, detail=True, **kwargs):
+    @classmethod
+    def _strip_protocol(cls, path):
+        """ Turn path from fully-qualified to file-system-specific
+        May require FS-specific handling, e.g., for relative paths or links.
+        """
+        path = stringify_path(path)
+        protos = (cls.protocol,) if isinstance(cls.protocol, str) else cls.protocol # NOQA
+        for protocol in protos:
```

Comment:
- What are the cases we get the protocol in front?
- I'd like to see some tests for this
- IMO easier done with a regular expression
```
path = re.sub('[a-z0-9]+:(//)?(.*)', r'\2', path)
return '/' + path if path[:0] != '/' else path
```

Reply:
Works if the path is passed as a string.  If it's passed as a pathlib object, will this work?  At any rate, the code below follows the same convention as what already exists in the fsspec library.